### PR TITLE
Migrating to .NET 8

### PR DESCRIPTION
**Pull request by david-staniec-octopus**
# Background

Due to .NET 6 going EOL on 12th November 2024, this PR removes .NET 6 as a target framework across Tentacle projects, and upgrades build scripts to use .NET 8.

MQS Handover Discussion

- Make sure all .net 6 references have been replaced with .net8 inside the repo
- Make sure not installing .net6 or .net8 where we don't need to (in the build shell script)
- Review / update all teamcity configuration to ensure we are not calling .net6 test / test names anymore
- Test build in Teamcity on a new / cloned build chain configuration to ensure the changes work as expected prior to us merging this PR to main
- Investigate and resolve cross platform build issue - referencing octo cli on .net6
- Test branch build of Tentacle in Octopus Server
- Should we create a significant new version of tentacle for .net8?

[SC-95888]

# How to review this PR

Quality :heavy_check_mark:

# Pre-requisites

- [ ] I have read [How we use GitHub Issues](https://github.com/OctopusDeploy/Issues/blob/main/docs/CONTRIBUTING.internal.md) for help deciding when and where it's appropriate to make an issue.
- [ ] I have considered informing or consulting the right people, according to the [ownership map](https://whimsical.com/ownership-map-NzbiD4HJyvhC9jNJNfS6TG).
- [ ] I have considered appropriate testing for my change.

### CODE REVIEW

File: source/Tentacle.sln.DotSettings
Comment (david-staniec-octopus) [inline source/Tentacle.sln.DotSettings:186]: Will remove this change

File: .gitignore
Comment (rhysparry) [inline .gitignore:59]: Nuke installs dotnet in this folder

File: build.ps1
Comment (rhysparry) [inline build.ps1:74]: We no longer need to install the .NET 6 runtime

File: build.sh
Comment (rhysparry) [inline build.sh:61]: We no longer need to install the .NET 6 runtime

File: build/Build.Pack.cs
Comment (rhysparry) [inline build/Build.Pack.cs:541]: The Octo CLI used by `OctopusTasks` is the old Octopus CLI, which requires .NET6 and is no longer maintained.

Consequently, this has been updated to use the new CLI.

File: build/Build.Pack.cs
Comment (rhysparry) [inline build/Build.Pack.cs:655]: Currently using a fixed version of the CLI. This gives some assurances regarding consistency and removes any complexities that arise from having to identify the current version.

It can be updated easily, although we may wish to make this constant more visible.

File: build/Build.Pack.cs
Comment (rhysparry) [inline build/Build.Pack.cs:681]: These are the only architectures we build the CLI for.

File: docker/linux/Dockerfile
Comment (rhysparry) [inline docker/linux/Dockerfile:13]: This uses the same original base OS image with the dependencies installed in the same manner as the Microsoft image.

Shifting away from the Microsoft image as the .NET6 EOL puts that image at high risk of being removed/out of date.

File: source/Octopus.Tentacle.Contracts/Octopus.Tentacle.Contracts.csproj
Comment (rhysparry) [inline source/Octopus.Tentacle.Contracts/Octopus.Tentacle.Contracts.csproj:31]: Note: Will need to update this to actual Halibut release once that has been merged/built.

File: source/Octopus.Tentacle.Tests.Integration/Support/TentacleTestCaseSourceAttribute.cs
Comment (rhysparry) [inline source/Octopus.Tentacle.Tests.Integration/Support/TentacleTestCaseSourceAttribute.cs:46]: This category is preserved as it is reference in TeamCity build configurations. Once this has merged, the build configuration can be updated and a subsequent PR can remove the obsolete category.

File: build/Build.Pack.cs
Comment (gb-8) [inline build/Build.Pack.cs:703]: Looks like this check should be done before we try to make the file executable.

File: source/Octopus.Tentacle.Tests.Integration/Support/TentacleTestCaseSourceAttribute.cs
Comment (gb-8) [inline source/Octopus.Tentacle.Tests.Integration/Support/TentacleTestCaseSourceAttribute.cs:173]: Can we remove this line already?

File: source/Tentacle.sln
Comment (gb-8) [inline source/Tentacle.sln:10]: I'm guess these lines are no longer required and are left over from previous changes?

File: source/Tentacle.sln
Comment (gb-8) [inline source/Tentacle.sln:70]: Just verifying these were redundant so don't need replacing with net8.0 ones?

**Review by gb-8 — APPROVED** [review]
Approved!
A few very minor comments to look at.

File: build/Build.Pack.cs
Comment (rhysparry) [reply build/Build.Pack.cs:703]: Sure. I think I kept the assert by the return as a last line of defence if things went wrong (we actually build on Windows, so `chmod` isn't needed but it meant I could easily test this path locally).

I'll update, because at least that will give a consistent assertion message in both cases.

File: source/Octopus.Tentacle.Tests.Integration/Support/TentacleTestCaseSourceAttribute.cs
Comment (rhysparry) [reply source/Octopus.Tentacle.Tests.Integration/Support/TentacleTestCaseSourceAttribute.cs:173]: Not quite yet. Old build configurations still reference the net6 category. Once this has merged, the build configuration can switch to the net8 category and then we can remove it.

File: source/Tentacle.sln
Comment (rhysparry) [reply source/Tentacle.sln:10]: Yeah, they stuck out as problems when I was using the Solution Items section to access the other files. Just some cleanup.

File: source/Tentacle.sln
Comment (rhysparry) [reply source/Tentacle.sln:70]: We already have net8.0 equivalents. This removes the net6.0 ones.

**Review by rain-on — APPROVED** [review]
Code owner approval of changes to Octopus.Tentacle.Kubernetes.Tsts.Integration.csproj.

**Pull request merged**
